lib/travis/api/v3/billing.rb

```diff
@@ -0,0 +1,56 @@
+module Travis::API::V3
+  class Billing
+    attr_reader :user_id, :subscription_id
+
+    def initialize(user_id = nil, subscription_id = nil)
+      @user_id = user_id
+      @subscription_id   = subscription_id
+    end
+
+    def create_subscription(subscription_params)
+      post('/subscriptions', {subscription: subscription_params})
+      #Billing will return a JSON representation of the created subscription
+    end
+
+    def cancel_subscription
+      post("/subscriptions/#{subscription_id}/cancel")
+    end
+
+    def edit_address(address)
+      patch("/subscriptions/#{subscription_id}/address", {address: address})
+    end
+
+    def patch(url, payload)
+      connect.patch do |req|
+        req.url url
+        req.headers['Content-Type'] = 'application/json'
+        req.body = payload
+      end
+    end
+
+    def post(url, payload)
+      connect.post do |req|
+        req.url url
+        req.headers['Content-Type'] = 'application/json'
+        req.body = payload
+      end
+    end
+
+    def connect
+      return unless billing_url && billing_auth_key
+      @connect ||= Faraday.new(billing_url, ssl: { ca_path: '/usr/lib/ssl/certs' }) do |client|
+        client.basic_auth 'admin', billing_auth_key
+        client.headers['X-Travis-User-Id'] = user_id
+        client.adapter :net_http
+      end
+    end
+
+    def billing_url
+      Travis.config.billing.url
```

Comment:
We still need to add this to the keychain - https://github.com/travis-pro/travis-pro-keychain/blob/master/config/travis.api.yml

and the Billing auth keys as ENV variables https://github.com/travis-ci/travis-api/pull/663/files#diff-55fb106cbe3f519c258881bde3303928R53
The Auth keys are already on Billing v2 keychain https://github.com/travis-pro/travis-pro-keychain/blob/master/config/travis.billing-v2.yml#L10